oci/table_oci_devops_project.go

```diff
@@ -0,0 +1,314 @@
+package oci
+
+import (
+	"context"
+	"strings"
+
+	oci_common "github.com/oracle/oci-go-sdk/v65/common"
+	"github.com/oracle/oci-go-sdk/v65/devops"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/turbot/go-kit/types"
+
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableDevopsProject(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:             "oci_devops_project",
+		Description:      "OCI Devops Project",
+		DefaultTransform: transform.FromCamel(),
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("id"),
+			Hydrate:    getProject,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listProjects,
+			KeyColumns: []*plugin.KeyColumn{
+				{
+					Name:    "compartment_id",
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "name",
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "id",
+					Require: plugin.Optional,
+				},
+			},
+		},
+		GetMatrixItemFunc: BuildCompartementRegionList,
+		Columns: []*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The name of the project.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "description",
+				Description: "The description of the project.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "id",
+				Description: "The OCID of the project.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "time_created",
+				Description: "The date and time the queue was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeCreated.Time"),
+			},
+			{
+				Name:        "time_updated",
+				Description: "The date and time the queue was last modified.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeUpdated.Time"),
+			},
+
+			{
+				Name:        "lifecycle_details",
+				Description: "A message describing the current state in more detail. For example, can be used to provide actionable information for a resource in Failed state.",
+				Type:        proto.ColumnType_STRING,
+			},
+
+			{
+				Name:        "lifecycle_state",
+				Description: "The current state of the project.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "namespace",
+				Description: "Namespace associated with the project.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "notification_topic_id",
+				Description: "The topic ID for the topic where project notifications will be published to.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.From(transformNotificationTopic),
```

Comment:
We can easily transform from json objects using the `transform.FromField` function

```suggestion
				Transform:   transform.FromField('NotificationConfig.TopicId'),
```